cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -0,0 +1,218 @@
+	}
+	if err = rows.Err(); err != nil {
+		log.Fatalf("error reading selected rows: %s", err)
+	}
+	return votes
+}
+
+// currentTotals returns a templateData structure for populating the web page.
+func currentTotals() templateData {
+
+	// get total votes for each candidate
+	var tabVotes, spaceVotes uint
+	_ = db.QueryRow(`select count(vote_id) from votes where candidate='TABS'`).Scan(&tabVotes)
+	_ = db.QueryRow(`select count(vote_id) from votes where candidate='SPACES'`).Scan(&spaceVotes)
+
+	// voteMargin is string representation of current voting margin.
+	voteDiff := int(math.Abs(float64(tabVotes) - float64(spaceVotes)))
+	var voteMargin string
+	if voteDiff == 1 {
+		voteMargin = "1 vote"
+	}
+	voteMargin = strconv.Itoa(voteDiff) + " votes"
+
+	return templateData{tabVotes, spaceVotes, voteMargin, recentVotes()}
+
+}
+
+// showTotals renders an HTML template showing the current vote totals.
+func showTotals(w http.ResponseWriter, r *http.Request) {
+	t, err := template.ParseFiles("templates/index.html")
+	if err != nil {
+		log.Fatalf("unable to parse template file: %s", err)
+	}
+
+	err = t.Execute(w, currentTotals())
+	if err != nil {
+		log.Fatalf("unable to execute template: %s", err)
+	}
+}
+
+// saveVote handles POST requests and saves a vote passed as http.Request form data.
+func saveVote(w http.ResponseWriter, r *http.Request) {
+	r.ParseForm()
+	team := r.Form["team"][0]
+	// [START cloud_sql_mysql_database-sql_connection]
+	sqlInsert := "INSERT INTO votes (candidate) VALUES (?)"
+	if team == "TABS" || team == "SPACES" {
+		if _, err := db.Exec(sqlInsert, team); err != nil {
+			log.Fatalf("unable to save vote: %s", err)
+		} else {
+			fmt.Fprintf(w, "Vote successfully cast for %s!\n", team)
+		}
+	}
+	// [END cloud_sql_mysql_database-sql_connection]
+}
+
+// mustGetEnv is a helper function for getting environment variables.
+// Displays a warning if the environment variable is not set.
+func mustGetenv(k string) string {
+	v := os.Getenv(k)
+	if v == "" {
+		fmt.Printf("Warning: %s environment variable not set.\n", k)
+	}
+	return v
+}
+
+// [START cloud_sql_mysql_database-sql_create]
+
+// initConnectionPool initializes a database connection for a Cloud SQL instance.
+func initConnectionPool() (*sql.DB, error) {
+	var (
+		dbUser                 = mustGetenv("DB_USER")
+		dbPass                 = mustGetenv("DB_PASS")
+		instanceConnectionName = mustGetenv("INSTANCE_CONNECTION_NAME")
+		dbName                 = mustGetenv("DB_NAME")
+	)
+
+	// Default connection string format for App Engine deployment (UNIX socket syntax).
+	dbURI := fmt.Sprintf("%s:%s@unix(/cloudsql/%s)/%s", dbUser, dbPass, instanceConnectionName, dbName)
+	if runtime.GOOS == "windows" {
+		// If running on Windows (local dev machine), connect via Cloud SQL Proxy.
+		instanceConnectionName = "127.0.0.1:3306"
+		dbURI = fmt.Sprintf("%s:%s@tcp(%s)/%s", dbUser, dbPass, instanceConnectionName, dbName)
+	}
+
+	dbConn, err := sql.Open("mysql", dbURI)
```

Comment:
This is actually a pool of connections -- we should make that clear in the naming
```suggestion
	dbPool, err := sql.Open("mysql", dbURI)
```